transaction/narayana/src/main/java/io/helidon/transaction/narayana/CdiExtension.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2025 Oracle and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.transaction.narayana;
+
+import java.util.Set;
+
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.spi.Extension;
+import jakarta.enterprise.inject.spi.ProcessAnnotatedType;
+
+/**
+ * Remove CDI beans registered by Narayana to allow registering them from Helidon service registry.
```

Comment:
There are much simpler ways to accomplish this. When #10313 goes in we may not need this class or any such facility.

Reply:
Well, it did not exist when I was implementing this code. :)